index.js

```diff
@@ -54,18 +68,204 @@ client.on("messageReactionAdd", async (reaction, user) => {
+        emojiReactToSend(reaction.message, user, TODO_TARGET_CHANNEL_ID);
+    }
+
+    // Check emojis for deleting todo:
+    const message = reaction.message.content;
+
+    if (reaction.emoji.name === DELETE_TODO_EMOJI) {
+        if (message.match(USERNAME_REGEX)[0] === `<@${user.id}>`) {
+            deleteTodo[0] = true;
+        }
+        if (message.match(USERNAME_REGEX)[1] === `<@${user.id}>`) {
+            deleteTodo[1] = true;
+        }
+        if (deleteTodo[0] === true && deleteTodo[1] === true) {
+            reaction.message.delete();
+            deleteTodo[0] = false;
+            deleteTodo[1] = false;
+        }
+    }
+});
+
+// Event listener for reactions removed from messages:
+client.on("messageReactionRemove", async (reaction, user) => {
+    // Ignore bot reactions:
+    if (user.bot) {
+        return;
+    }
+
+    // Fetch partials if necessary:
+    if (reaction.partial) {
+        try {
+            await reaction.fetch();
+        }
+        catch (error) {
+            console.error("Something went wrong when fetching the reaction: ", error);
+            return;
+        }
+    }
+
+    // Check emojis removed for deleting todo:
+    const message = reaction.message.content;
+
+    if (message.match(USERNAME_REGEX)[0] === `<@${user.id}>`) {
+        deleteTodo[0] = false;
+    }
+    if (message.match(USERNAME_REGEX)[1] === `<@${user.id}>`) {
+        deleteTodo[1] = false;
+    }
+});
+
+// Event listener for messages sent:
+client.on("messageCreate", async (message) => {
+    // Skip bot's own messages:
+    if (message.author.bot) {
+        return;
+    }
+
+
+    // Check if message is a reply:
+    if (message.reference) {
+        const originalMessageId = message.reference.messageId;
+
+        try {
+            // Fetch original message being replied to:
+            const originalMessage = await message.channel.messages.fetch(originalMessageId);
+            const originalMessageLink = `https://discord.com/channels/${originalMessage.guildId}/${originalMessage.channelId}/${originalMessage.id}`;
+
+            // Add todo:
+            if (message.content.substring(0, 5).match(TODO_REGEX)) {
+                sendTodo(originalMessage, message.author, TODO_TARGET_CHANNEL_ID);
+            }
+            else if (message.content.match(TODO_REGEX)) {
+                sendTodo(message, message.author, TODO_TARGET_CHANNEL_ID);
+            }
+
+            // Check if original message meets time threshold:
+            const messageAgeInDays = (Date.now() - originalMessage.createdTimestamp) / TIME_IN_DAYS;
+            const ageThresholdInDays = 7;
+
+            if (messageAgeInDays > ageThresholdInDays) {
+                //Include snippet of original message for context:
+                
+                let snippet = originalMessage.content;
+
+                if (originalMessage.content.length > CHAR_LENGTH) {
+                    const spaceIndex = originalMessage.content.indexOf(" ", 500);
```

Comment:
500 is an arbitrary constant and can be extracted to the constants in the file

Reply:
I forgot to replace that one. Will change in PR